src/artefacts_demo_control/artefacts_demo_control/collision_test.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+import sys
+import os 
+import subprocess
+import json
+import re
+from time import sleep
+
+import pytest
+import launch_testing
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch_ros.actions import Node
+
+command = ["gz", "model", "-m", "artefacts_box", "-i"]
+result = subprocess.run(command, check = True, text = True, stdout=subprocess.PIPE)
+result_output = result.stdout
+
+# Extract position data if it exists
+position_pattern = r"pose\s*{\s*position\s*{\s*x:\s*(-?\d+\.\d+)\s*y:\s*(-?\d+\.\d+)\s*z:\s*(-?\d+\.\d+)"
+
+# Search for the x, y, and z variables using the regular expression
+match = re.search(position_pattern, result_output, re.DOTALL)
+
+# Extract the x, y, and z variables if a match is found
+if match:
+    x_variable = float(match.group(1))
+    y_variable = float(match.group(2))
+    z_variable = float(match.group(3))
+    print(f"x: {x_variable}")
+    print(f"y: {y_variable}")
+    print(f"z: {z_variable}")
+else:
+    print("No x, y, or z variables found in the text.")
+
+# Command for running node as a subprocess
+command_node = ["ros2", "run", "artefacts_demo_control", "artefacts_control"]
+running_node = subprocess.run(command_node, check = True, text = True, stdout=subprocess.PIPE)
```

Comment:
@MDecarabas it looks to me like the process is now starting twice.
Onece here, and once here, and once via the Execute process in the launch description.
Is this line needed?

Reply:
Yes, it does technically start twice. The one inside the node block errors out and fails. However, if that line is not there the test fails to execute

Reply:
> . The one inside the node block errors out and fails. However, if that line is not there the test fails to execute

I am confused, yesterday, you mentioned that it was starting after you switched to ExecuteProcess. what makes it fail now?

Reply:
rechecking the thread, you mention that it started failing after using launch_testing.actions.ReadyToTest()
=> So maybe the Node based version was starting straight away, while theExecute process was delaying the start until the ReadyToTest,

Reply:
I figured out what the problem was. The node was starting but instantly the test was forcing a SIGINT. I believe it worked when my machine was slow to send that command and I saw something execute. Once sleeps were implemented that aspect was fixed.

Reply:
You say that it was fixed, but it still looks like it is starting twice, no?

Reply:
This is the old version I left in case it might be better to use